pkgs/development/python-modules/jaxlib/default.nix

```diff
@@ -204,40 +213,60 @@ let
 
     removeRulesCC = false;
 
-    GCC_HOST_COMPILER_PREFIX = lib.optionalString cudaSupport "${cudatoolkit_cc_joined}/bin";
-    GCC_HOST_COMPILER_PATH = lib.optionalString cudaSupport "${cudatoolkit_cc_joined}/bin/gcc";
+    GCC_HOST_COMPILER_PREFIX = lib.optionalString cudaSupport "${backend_cc_joined}/bin";
+    GCC_HOST_COMPILER_PATH = lib.optionalString cudaSupport "${backend_cc_joined}/bin/gcc";
 
     # The version is automatically set to ".dev" if this variable is not set.
     # https://github.com/google/jax/commit/e01f2617b85c5bdffc5ffb60b3d8d8ca9519a1f3
     JAXLIB_RELEASE = "1";
 
-    preConfigure = ''
+    preConfigure =
       # dummy ldconfig
-      mkdir dummy-ldconfig
-      echo "#!${stdenv.shell}" > dummy-ldconfig/ldconfig
-      chmod +x dummy-ldconfig/ldconfig
-      export PATH="$PWD/dummy-ldconfig:$PATH"
-      cat <<CFG > ./.jax_configure.bazelrc
-      build --strategy=Genrule=standalone
-      build --repo_env PYTHON_BIN_PATH="${python}/bin/python"
-      build --action_env=PYENV_ROOT
-      build --python_path="${python}/bin/python"
-      build --distinct_host_configuration=false
-      build --define PROTOBUF_INCLUDE_PATH="${pkgs.protobuf}/include"
-    '' + lib.optionalString (stdenv.hostPlatform.avxSupport && stdenv.hostPlatform.isUnix) ''
-      build --config=avx_posix
-    '' + lib.optionalString mklSupport ''
-      build --config=mkl_open_source_only
-    '' + lib.optionalString cudaSupport ''
-      build --action_env CUDA_TOOLKIT_PATH="${cudatoolkit_joined}"
-      build --action_env CUDNN_INSTALL_PATH="${cudnn}"
-      build --action_env TF_CUDA_PATHS="${cudatoolkit_joined},${cudnn},${nccl}"
-      build --action_env TF_CUDA_VERSION="${lib.versions.majorMinor cudatoolkit.version}"
-      build --action_env TF_CUDNN_VERSION="${lib.versions.major cudnn.version}"
-      build:cuda --action_env TF_CUDA_COMPUTE_CAPABILITIES="${builtins.concatStringsSep "," cudaFlags.realArches}"
-    '' + ''
-      CFG
-    '';
+      # TODO: test whether we still need this
+      ''
+        mkdir dummy-ldconfig
+        echo "#!${backendStdenv.shell}" > dummy-ldconfig/ldconfig
```

Comment:
@SomeoneSerge I'm assuming that this ought to be `backendStdenv.shell` instead of `stdenv.shell` but I'm not really sure. There are also a number of other instances where `stdenv` is used in the derivation... not sure which of those, if any, we should switch over?

Reply:
1. > assuming that this ought to be backendStdenv.shell instead of stdenv.shell

   in principle, this shouldn't matter, I think they refer to the same shell in practice
2. > There are also a number of other instances where stdenv is used in the derivation

   ...would be nice to be consistent, e.g. the way our opencv derivation does it (defines the `effectiveStdenv`, and hides the old `stdenv` with the new `throw` definition)
3. ...but would also be nice to kill the `backendStdenv` off

Reply:
ahhh tough choices! ok i'll plan on going the `effectiveStdenv` route, if that sounds acceptable to you? we can always remove `backendStdenv` later once the path to doing so is clear

Reply:
Yes, when we remove `backendStdenv` we remove it treewide. It's not clear to me how to do that yet

Reply:
done in https://github.com/NixOS/nixpkgs/pull/288857/commits/c0e2764852da93be1ee5b6f1d199bcced94385c2